src/module/data/pseudo-documents/pseudo-document.mjs

```diff
@@ -111,6 +112,16 @@ export default class PseudoDocument extends foundry.abstract.DataModel {
 
   /* -------------------------------------------------- */
 
+  /**
+   * Fetches the collection this PseudoDocument is contained in.
+   * @type {ModelCollection<this>}
+   */
+  get collection() {
+    return foundry.utils.getProperty(this.document, this.fieldPath);
```

Comment:
I don't think this will work in case of nested pseudo documents, and it would also be better to use the `getEmbeddedPseudoDocumentCollection` method.

Better yet, we make this a getter for a private property instantiated when the pseudo document is initialized. We do this already for the model, I think.